lib/testing/integration/integration.go

```diff
@@ -264,7 +264,7 @@ func (integration *Integration) Version() Version {
 }
 
 // NewAuthService creates a new auth server instance.
-func (integration *Integration) NewAuthService() (*AuthService, error) {
+func (integration *Integration) NewAuthService(cacheEnabled bool) (*AuthService, error) {
```

Comment:
Instead of passing bool flag, can you make this into a functional "option" argument? Something like

```go
type AuthServiceOption func(yaml string) string

func WithCache() AuthServiceOption {
  return func(yaml string) string {
    return strings.Replace(yaml, "{{TELEPORT_CACHE_ENABLED}}, "true")
  }
}

func (integration *Integration) NewAuthService(opts ...AuthServiceOption) (*AuthService, error) { ... }
```

Then you won't need to modify existing tests also, or explain what passing "true" or "false" means.

Reply:
That's awesome, thank you